core/src/main/scala/water/support/ModelSerializationSupport.scala

```diff
@@ -54,10 +56,23 @@ trait ModelSerializationSupport {
     destination
   }
 
-  def exportMOJOModel(model: Model[_, _, _], destination: URI): URI = {
-    val destFile = new File(destination)
-    val fos = new FileOutputStream(destFile)
-    model.getMojo.writeTo(fos)
+  def exportMOJOModel(model: Model[_, _, _], destination: URI, sc: SparkContext): URI = {
+    if (destination.getScheme() == "hdfs") {
+      try {
+        val fs = FileSystem.get(sc.hadoopConfiguration)
+        val output = fs.create(new Path(destination))
+        val os = new BufferedOutputStream(output)
+        model.getMojo.writeTo(os)
+      } catch {
+        case e: IllegalArgumentException =>
+          throw new IllegalArgumentException (s"Error while writing to HDFS " +
+            s"(most likely no HDFS is available) in destination $destination : ${e.getMessage}")
+      }
+    } else {
+      val destFile = new File(destination)
+      val fos = new FileOutputStream(destFile)
+      model.getMojo.writeTo(fos)
```

Comment:
One more comment: we should probably have one more abstraction:
```
def exportModel(model, destination: OutputStream) {
...
}
```

The method `exportModel` would just created OS based on schema and pass to function ^.

Furthermore, the output stream handling should be handled by our Persist layer.

Reply:
Hi @mmalohlava,

Thanks for your feedback. 

do you mean a function that will return an OutputStream? something like that?

def exportModel(model, destination: URI):  OutputStream{
...
}

Regarding the second point, do you mean the exception handling? should I remove try/catch part?

Reply:
By Persistent storage we mean this code
```
    val p: Persist = H2O.getPM.getPersistForURI(destinationURI)
    val os: OutputStream = p.create(destination.toString, true)
```

It's used also in the `exportH2OModel` method, so you can have a look for inspiration.

Reply:
@mmalohlava I'm a bit lost what you mean by `exportModel` method. We need to distinguish the cases for exporting MOJO, POJO and binary model.